src/overlays/gamestates/ovl_debug_opening/z_debug_opening.c

```diff
@@ -0,0 +1,316 @@
+    this->state.running = false;
+    SET_NEXT_GAMESTATE(&this->state, Play_Init, PlayState);
+}
+
+void DebugOpening_LoadTitleScreen(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    gSaveContext.seqId = (u8)NA_BGM_DISABLED;
+    gSaveContext.natureAmbienceId = 0xFF;
+    gSaveContext.gameMode = GAMEMODE_TITLE_SCREEN;
+    this->state.running = false;
+    SET_NEXT_GAMESTATE(&this->state, TitleSetup_Init, TitleSetupState);
+}
+
+void DebugOpening_LoadFileSelect(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    gSaveContext.gameMode = GAMEMODE_FILE_SELECT;
+    SET_NEXT_GAMESTATE(&this->state, FileSelect_Init, FileSelectState);
+    this->state.running = false;
+}
+
+void DebugOpening_LoadMapSelect(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    gSaveContext.gameMode = GAMEMODE_NORMAL;
+    SET_NEXT_GAMESTATE(&this->state, MapSelect_Init, MapSelectState);
+    this->state.running = false;
+}
+
+/**
+ * sOptionInfo contains every usable option in the first page.
+ * The .name of every option is printed in DebugOpening_DrawOptions
+ * through a loop and the .func gets executed based on this->currentOption
+ * in DebugOpening_ControlOptions
+ */
+static OptionInfo sOptionInfo[] = {
+    { .func = DebugOpening_ChooseSaveFile, .name = "Save File: " },
+    {
+        .func = DebugOpening_LoadDefinedScene,
+        .name = "Init Defined Scene",
+    },
+    {
+        .func = DebugOpening_LoadMapSelect,
+        .name = "Init Map Select",
+    },
+    {
+        .func = DebugOpening_LoadTitleScreen,
+        .name = "Init Title Screen",
+    },
+    {
+        .func = DebugOpening_LoadFileSelect,
+        .name = "Init File Select",
+    },
+};
+
+void DebugOpening_ControlOptions(GameState* thisx) {
+    DebugOpeningState* this = (DebugOpeningState*)thisx;
+
+    // if dpad-up is pressed, go up in the options page
+    if (CHECK_BTN_ANY(this->state.input[0].press.button, BTN_DUP) && this->currentOption > OPTION_CHOOSE_SAVE_FILE) {
+        this->currentOption--;
+        Audio_PlaySfxGeneral(NA_SE_IT_SWORD_IMPACT, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
+                             &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+    }
+
+    // if dpad-down is pressed, go down in the options page
+    if (CHECK_BTN_ANY(this->state.input[0].press.button, BTN_DDOWN) && this->currentOption < OPTION_LOAD_FILE_SELECT) {
+        this->currentOption++;
+        Audio_PlaySfxGeneral(NA_SE_IT_SWORD_IMPACT, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
+                             &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+    }
+
+    // Run the function of the currently selected option
+    // as soon as A is pressed
+    if (CHECK_BTN_ALL(this->state.input[0].press.button, BTN_A)) {
+        sOptionInfo[this->currentOption].func(thisx);
+    }
+}
+
+#define STRINGIFY(s) #s
+#define EXPAND_AND_STRINGIFY(s) STRINGIFY(s)
```

Comment:
I wonder if it's worth moving these to macros.h, so it's not defined twice (the other one being in build.c I think?)